dockstore-webservice/src/main/java/io/dockstore/webservice/core/Validation.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *    Copyright 2018 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.webservice.core;
+
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.ComparisonChain;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+import org.json.JSONObject;
+
+/**
+ * This describes the validation information associated with one or more files for a version
+ * @author aduncan
+ * @since 1.6.0
+ */
+@ApiModel("Validation")
+@Entity
+@Table(name = "validation")
+@SuppressWarnings("checkstyle:magicnumber")
+public class Validation implements Comparable<Validation> {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @ApiModelProperty(value = "Implementation specific ID for the source file in this web service", required = true, position = 0)
+    private long id;
+
+    @Enumerated(EnumType.STRING)
+    @ApiModelProperty(value = "Enumerates the type of file", required = true, position = 1)
+    private SourceFile.FileType type;
+
+    @Column
+    @ApiModelProperty(value = "Is the file type valid", required = true, position = 2)
+    private Boolean valid = false;
```

Comment:
We probably don't actually want true, false, null logic here. So a "boolean" would probably be better.